osu.Game/Overlays/SettingsToolboxGroup.cs

```diff
@@ -132,6 +134,16 @@ protected SettingsToolboxGroup(string title)
         private const float fade_duration = 800;
         private const float inactive_alpha = 0.5f;
 
+        protected override bool OnInvalidate(Invalidation invalidation, InvalidationSource source)
+        {
+            // These toolbox grouped may be contracted to only show icons.
+            // For now, let's hide the header to avoid text truncation weirdness in such cases.
+            if (invalidation.HasFlagFast(Invalidation.DrawSize))
+                headerText.FadeTo(headerText.DrawWidth < DrawWidth ? 1 : 0, 150, Easing.OutQuint);
```

Comment:
I'm fuzzy on the details of that invalidation subsystem again, but wasn't the point of `OnInvalidate()` that at that point layout and the properties related to it have invalid values at that point and therefore that `headerText.DrawWidth < DrawWidth` check shouldn't/can't occur there? As in `OnInvalidate()` implementations should only mark that there's a change coming, but the actual re-validation/re-layout should happen on the next `Update()` call?

Reply:
This is correct, `OnInvalidate` should never refer to `DrawWidth`.